sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -624,67 +624,89 @@ object NullPropagation extends Rule[LogicalPlan] {
  */
 object FoldablePropagation extends Rule[LogicalPlan] {
   def apply(plan: LogicalPlan): LogicalPlan = {
-    var foldableMap = AttributeMap(plan.flatMap {
-      case Project(projectList, _) => projectList.collect {
```

Comment:
How about Aggregate? Why we are not handling Aggregate in the original rule?

Reply:
This ticket and PR just focuses on the correctness issue found. But I think we can extend the rule to handle `Aggregate` in another ticket if that is ok with you.

Reply:
I had the same question when I first looked at the PR, but since this PR might need to be hotfixed to released branches, it's a good idea to keep it simple and don't add any new features. So I'm +1 for handling Aggregate in a separate PR.

Reply:
+1 for @peter-toth and @rednaxelafx 's opinion.